.github/workflows/checks.yaml

```diff
@@ -58,7 +60,15 @@ jobs:
            --header 'Accept: application/json' \
            "${PROBE_ENDPOINT}" \
           ) == '200' || exit 1
-    - name: Install pytest
-      run: pip install pytest
-    - name: Test Endpoints with pytest (Integration Tests)
-      run: pytest tests/test_endpoints.py
+    - name: Calculate integration test coverage
+      run: |
+        coverage run -m pytest tests/integrationTest/test_endpoints.py -W ignore::DeprecationWarning
```

Comment:
You are calculating the coverage of your _test_ code, not the actual application code. This doesn't make sense.

Btw, in your PR description, I have seen from your unit test coverage report that you also include the test code (along with the application code). I have removed the modules from the test code and have re-calculated the coverage (from 91% to 88%) for the application code only.

Have a look at the `--source` parameter to tell `coverage` which application code to test. Of course you also need to make sure that only the correct tests (unit _or_ integration) are ran.

Reply:
But in the case of integration testing, it will not check the application code. What you told me will be applicable for unit testing and I have not added unit tests here(due to the error). Moreover, if I do add --source pro_tes in the coverage in case of integration testing it will show 0 coverage. as integration tests don't actually go through all the application code. It will just check that deployed application is returning the expected output.

Reply:
Okay. But in that case it also doesn't make sense to compute the coverage for it. Certainly we are not interested in the coverage of the test code.

So please simply remove the coverage from the CI and we should be good to go.